controllers/storagecluster/external_resources_test.go

```diff
@@ -474,3 +515,82 @@ func assertReconciliationOfExternalResource(t *testing.T, reconciler StorageClus
 	assert.Equal(t, secondExtSecretChecksum, thirdExtSecretChecksum)
 
 }
+
+func TestExternalMonitoringResources(t *testing.T) {
+	type testResources struct {
+		ExternalResource
+		Label                   string
+		ReconcileExpectedToFail bool
+	}
+	monAddedExternalResources := []testResources{
+		{
+			ExternalResource: ExternalResource{
+				Kind: "CephCluster",
+				Data: map[string]string{
+					"MonitoringEndpoint": "127.0.0.1",
+					"MonitoringPort":     fmt.Sprint(generateRandomPort(30000, 40000)),
+				},
+				Name: "monitoring-endpoint",
+			},
+			Label:                   "Case 1",
+			ReconcileExpectedToFail: false,
+		},
+		{
+			ExternalResource: ExternalResource{
+				Kind: "CephCluster",
+				Data: map[string]string{
+					"MonitoringEndpoint": "127.0.0.1",
+					"MonitoringPort":     fmt.Sprint(generateRandomPort(30000, 40000)),
+				},
+				Name: "monitoring-endpoint",
+			},
+			Label:                   "Case 2",
+			ReconcileExpectedToFail: false,
+		},
```

Comment:
How is this different from the above case (other than the fact that you're generating random port)?

Reply:
They are the same, I was adding two passing scenarios... As it may feel redundant, removed it